### ISSUE TITLE
SSH private key passphrase support

### ISSUE THREAD

**Issue by idahood** [netenglabs/suzieq#238]
**Is your feature request related to a problem? Please describe.**
I would like to use my personal SSH key to authenticate connections to network devices. This key has a passphrase. If I mount the key into the sq-poller image via `-v ~/.ssh/id_rsa:/root/.ssh/id_rsa` I see the following error in `/tmp/sq-poller.log`:

```
2020-08-24 14:40:43,655 - suzieq.poller.nodes.node - ERROR - ERROR: Unable to read private key file /root/.ssh/id_rsa for 192.168.3.235 due to Passphrase must be specified to import encrypted private keys
```

**Describe the solution you'd like**
Ideally I would like to re-use the OpenSSH configuration files/ecosystem for SSH based authentication in SuzieQ. 

asycssh appears to have support for reading keys from ssh-agent:
https://asyncssh.readthedocs.io/en/stable/_modules/asyncssh/agent.html#SSHAgentClient.get_keys

I'd also take a less robust solution of providing the passphrase via a prompt/script if that's a quicker path to implementation.

**Additional context**

**Comment by ddutt** [comment]
Initial patch submitted via commit c39469f00b420e979fe8cf39cf30e5b98e2afbdc

User interface is:
* Invoke sq-poller with --passphrase option
* You'll be prompted to enter passphrase to decrypt private key file and the decrypted private key file is used to connect to devices.

**Comment by idahood** [comment]
Tested without the option and providing wrong credential
```
2020-08-27 14:38:04,804 - suzieq.poller.nodes.node - ERROR - ERROR: Unable to read private key file /root/.ssh/id_rsa for 192.168.3.242 due to Unable to decrypt PKCS#1 private key
```

I do not see that error line when providing the correct passphrase so this appears to be working. (I am unable to fully validate due to the ProxyJump issues that we discussed in private).

**Comment by ddutt** [comment]
Fixed with commit 4d6d5ede2f2bad4de94d143290f3fb755c671a72
